File: RB101_RB109_small_problems/easy_4/what_century_is_that.rb
SUFFIX = {
  1 => "st",
  2 => "nd",
  3 => "rd",
}

SUFFIX.default = "th"

def get_suffix(century)
  century_digits = century.digits
  if century_digits[1] == 1
    'th'
  else
    SUFFIX[century_digits[0]]
  end
end

def century(year)
  century = (year / 100.0).ceil
  "#{century}#{get_suffix(century)}"
end

p century(2000) == '20th'
p century(2001) == '21st'
p century(1965) == '20th'
p century(256) == '3rd'
p century(5) == '1st'
p century(10103) == '102nd'
p century(1052) == '11th'
p century(1127) == '12th'
p century(11201) == '113th'

File: JS210_small_problems/easy_2/what_century_is_that.js
function century(year) {
  let centuryNumber = Math.ceil(year / 100).toString();
  let suffix;
  
  if (centuryNumber.at(-2) === '1') {
    suffix = 'th';
  } else if (centuryNumber.at(-1) == '1') {
    suffix = 'st';
  } else if (centuryNumber.at(-1) === '2') {
    suffix = 'nd';
  } else if (centuryNumber.at(-1) === '3') {
    suffix = 'rd';
  } else {
    suffix = 'th';
  }
  
  return centuryNumber + suffix;
}

console.log(century(2000));        // "20th"
console.log(century(2001));        // "21st"
console.log(century(1965));        // "20th"
console.log(century(256));         // "3rd"
console.log(century(5));           // "1st"
console.log(century(10103));       // "102nd"
console.log(century(1052));        // "11th"
console.log(century(1127));        // "12th"
console.log(century(11201));       // "113th"